### DISCUSSION TITLE
possible to use on github page?

### DISCUSSION THREAD

**Discussion by lalrochhara** [colinwilson/lotusdocs#44] [Q&A] [answered]
hello, like my topic, can we use this script to github.io page? is it possible to use?
if yes, where can i found documentation or guide?

thanks in advance.

**Comment by colinwilson** [comment, marked as answer]
Absolutely! But I haven't completed the documentation regarding deployments on [lotusdocs.dev](https://lotusdocs.dev/docs/) yet.

You can use the following instructions as a guide to setting up Lotus Docs on GitHub pages:

1. Create a new repo on GitHub e.g. `https://github.com/lalrochhara/lalrochhara.github.io`
2. Follow the [quickstart guide](https://lotusdocs.dev/docs/quickstart/#create-a-new-lotus-docs-site) to create your new site locally.

       hugo new site lalrochhara.github.io && cd lalrochhara.github.io

3. Update your `hugo.toml` config to include the required theme modules and update your `baseURL` to your intended GitHub Pages domain e.g. `lalrochhara.github.io`. Also configure a `[[menu.primary]]` item to create a link to the `docs/` section on the home page.

        baseURL = 'https://lalrochhara.github.io'
        languageCode = 'en-us'
        title = 'My New Hugo Site'
        
        [module]
            [[module.imports]]
                path = "github.com/colinwilson/lotusdocs"
                disable = false
            [[module.imports]]
                path = "github.com/gohugoio/hugo-mod-bootstrap-scss/v5"
                disable = false
        
        [[menu.primary]]
            name  = "Docs"
            url = "/docs/"
            identifier = "docs"
            weight = 10

4. Create some example content
        
        hugo new docs/example-page.md

5. Update the newly created `example-page.md` file to `draft: false` (This is so it shows up when you build the Lotus Docs site)

        ---
        title: "Example Page"
        date: 2023-08-25T23:36:29+01:00
        draft: false
        ---

6. Commit and push your local repo to the one you created on GitHub.

7. Visit your GitHub repository. From the main menu choose Settings > Pages. In then center of your screen you will see this:
    
    ![](https://gohugo.io/hosting-and-deployment/hosting-on-github/gh-pages-1.png)

8. Change the Source to `GitHub Actions`. The change is immediate; you do not have to press a Save button.

    ![](https://gohugo.io/hosting-and-deployment/hosting-on-github/gh-pages-2.png)

9. Create an empty file in your local repository.

       .github/workflows/hugo.yaml

10. Copy and paste the YAML below into the file you created. Change the branch name and Hugo version as needed.
        
        # Sample workflow for building and deploying a Hugo site to GitHub Pages
        name: Deploy Hugo site to Pages
        
        on:
          # Runs on pushes targeting the default branch
          push:
            branches:
              - main
        
          # Allows you to run this workflow manually from the Actions tab
          workflow_dispatch:
        
        # Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
        permissions:
          contents: read
          pages: write
          id-token: write
        
        # Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
        # However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
        concurrency:
          group: "pages"
          cancel-in-progress: false
        
        # Default to bash
        defaults:
          run:
            shell: bash
        
        jobs:
          # Build job
          build:
            runs-on: ubuntu-latest
            env:
              HUGO_VERSION: 0.117.0
            steps:
              - name: Install Hugo CLI
                run: |
                  wget -O ${{ runner.temp }}/hugo.deb https://github.com/gohugoio/hugo/releases/download/v${HUGO_VERSION}/hugo_extended_${HUGO_VERSION}_linux-amd64.deb \
                  && sudo dpkg -i ${{ runner.temp }}/hugo.deb
              - name: Checkout
                uses: actions/checkout@v3
                with:
                  submodules: recursive
                  fetch-depth: 0
              - name: Setup Pages
                id: pages
                uses: actions/configure-pages@v3
              - name: Install Node.js dependencies
                run: "[[ -f package-lock.json || -f npm-shrinkwrap.json ]] && npm ci || true"
              - name: Build with Hugo
                env:
                  # For maximum backward compatibility with Hugo modules
                  HUGO_ENVIRONMENT: production
                  HUGO_ENV: production
                run: |
                  hugo \
                    --gc \
                    --minify \
                    --baseURL "${{ steps.pages.outputs.base_url }}/"
              - name: Upload artifact
                uses: actions/upload-pages-artifact@v1
                with:
                  path: ./public
        
          # Deployment job
          deploy:
            environment:
              name: github-pages
              url: ${{ steps.deployment.outputs.page_url }}
            runs-on: ubuntu-latest
            needs: build
            steps:
              - name: Deploy to GitHub Pages
                id: deployment
                uses: actions/deploy-pages@v2

11. Commit the change to your local repository with a commit message of something like “Add workflow”, and push to GitHub.
12.  From GitHub’s main menu, choose Actions. You will see something like this:

     ![](https://gohugo.io/hosting-and-deployment/hosting-on-github/gh-pages-3.png)

13.  When GitHub has finished building and deploying your site, the color of the status indicator will change to green.

     ![](https://gohugo.io/hosting-and-deployment/hosting-on-github/gh-pages-4.png)

14. Click on the commit message as shown above. You will see this:

     ![](https://gohugo.io/hosting-and-deployment/hosting-on-github/gh-pages-5.png)

    Under the deploy step, you will see a link to your live site.

    In the future, whenever you push a change from your local repository, GitHub will rebuild your site and deploy the changes.

**Reply by lalrochhara** [reply]
wow, thanks you so much, for your time and explain for me❤️
